utils/threshAlgos.py

```diff
@@ -169,19 +175,18 @@ def anaUltraThreshold(args,thrFilename,GEBtype="short",outputDir=None,fileScurve
         if args.isVFAT2:
             dict_trimRange[int(event.vfatN)] = int(event.dict_trimRange)
 
-        if not (array_chipID[event.vfatN] > 0):
-            if 'vfatID' in listOfBranches:
-                array_chipID[event.vfatN] = event.vfatID
-            else:
-                array_chipID[event.vfatN] = 0
-
         stripPinOrChan = dict_vfatChanLUT[event.vfatN][stripChanOrPinType][event.vfatCH]
         dict_h2D_thrDAC[event.vfatN].Fill(stripPinOrChan,event.vth1,event.Nhits)
         pass
 
-    # Make output TTree
+    # Make output TFile and make output TTree
     from array import array
+    outFile = r.TFile("{0}/{1}".format(outputDir,args.outfilename),"RECREATE")
```

Comment:
Why is this now needed?

Reply:
In order to avoid errors:

```
Error in <TTree::Fill>: Failed filling branch:thrAnaTree.detName, nbytes=-1, entry=1680
 This error is symptomatic of a Tree created as a memory-resident Tree
 Instead of doing:
    TTree *T = new TTree(...)
    TFile *f = new TFile(...)
 you should do:
    TFile *f = new TFile(...)
    TTree *T = new TTree(...)
```